### PR TITLE
Issue #2984 Make testing of app deployments more robust

### LINKED ISSUE #2984: Deploying sclorg/nodejs-ex through integration tests fails in CentOS CI
`oc new-app https://github.com/sclorg/nodejs-ex -l name=myapp` is failing in CI https://ci.centos.org/job/minishift-pr/5372/console

### General information

  * Minishift version: 1.26.1
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

Test scenario cmd-openshift.feature

```
Scenario: User deploys nodejs example application from OpenShift repository
    Given Minishift has state "Running"
     When executing "oc new-app https://github.com/sclorg/nodejs-ex -l name=myapp" retrying 20 times with wait period of 3 seconds
     Then stdout should contain
      """
      Run 'oc status' to view your app.
      """
```

### Steps to reproduce

  1.  Not able to reproduce locally. It is only seen in CentOS CI

### Expected
```
$ oc new-app https://github.com/sclorg/nodejs-ex -l name=myapp
--> Found image 93de123 (2 weeks old) in image stream "openshift/nodejs" under tag "10" for "nodejs"

    Node.js 10.12.0 
    --------------- 
    Node.js  available as docker container is a base platform for building and running various Node.js  applications and frameworks. Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient, perfect for data-intensive real-time applications that run across distributed devices.

    Tags: builder, nodejs, nodejs-10.12.0

    * The source repository appears to match: nodejs
    * A source build using source code from https://github.com/sclorg/nodejs-ex will be created
      * The resulting image will be pushed to image stream tag "nodejs-ex:latest"
      * Use 'start-build' to trigger a new build
    * This image will be deployed in deployment config "nodejs-ex"
    * Port 8080/tcp will be load balanced by service "nodejs-ex"
      * Other containers can access this service through the hostname "nodejs-ex"

--> Creating resources with label name=myapp ...
    imagestream.image.openshift.io "nodejs-ex" created
    buildconfig.build.openshift.io "nodejs-ex" created
    deploymentconfig.apps.openshift.io "nodejs-ex" created
    service "nodejs-ex" created
--> Success
    Build scheduled, use 'oc logs -f bc/nodejs-ex' to track its progress.
    Application is not exposed. You can expose services to the outside world by executing one or more of the commands below:
     'oc expose svc/nodejs-ex' 
    Run 'oc status' to view your app.
```

### Actual
```
--- Failed steps:
10:03:33 
10:03:33   Scenario: User deploys nodejs example application from OpenShift repository # features/cmd-openshift.feature:50
10:03:33     Then stdout should contain # features/cmd-openshift.feature:53
10:03:33       Error: Output did not match. Expected: 'Run 'oc status' to view your app.', Actual: '--> Found image 93de123 (2 weeks old) in image stream "openshift/nodejs" under tag "10" for "nodejs"
10:03:33 
10:03:33     Node.js 10.12.0 
10:03:33     --------------- 
10:03:33     Node.js  available as docker container is a base platform for building and running various Node.js  applications and frameworks. Node.js is a platform built on Chrome's JavaScript runtime for easily building fast, scalable network applications. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient, perfect for data-intensive real-time applications that run across distributed devices.
10:03:33 
10:03:33     Tags: builder, nodejs, nodejs-10.12.0
10:03:33 
10:03:33     * The source repository appears to match: nodejs
10:03:33     * A source build using source code from https://github.com/sclorg/nodejs-ex will be created
10:03:33       * The resulting image will be pushed to image stream tag "nodejs-ex:latest"
10:03:33       * Use 'start-build' to trigger a new build
10:03:33     * This image will be deployed in deployment config "nodejs-ex"
10:03:33     * Port 8080/tcp will be load balanced by service "nodejs-ex"
10:03:33       * Other containers can access this service through the hostname "nodejs-ex"
10:03:33 
10:03:33 --> Creating resources with label name=myapp ...
10:03:33 --> Failed
10:03:33 '
```

### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

### PR DESCRIPTION
**Pull request by agajdosi**
Fixes #2984.

### CODE REVIEW

**Comment by agajdosi** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
retest this please

File: test/integration/features/cmd-openshift.feature [hunk 1]
```diff
@@ -48,7 +48,7 @@ cluster in VM provided by Minishift.
 
   Scenario: User deploys nodejs example application from OpenShift repository
     Given Minishift has state "Running"
-     When executing "oc new-app https://github.com/sclorg/nodejs-ex -l name=myapp" succeeds
+     When executing "oc new-app https://github.com/sclorg/nodejs-ex -l name=myapp" retrying 20 times with wait period of "3s"
```
Comment (LalatenduMohanty) [inline test/integration/features/cmd-openshift.feature]: Interesting, I did not catch this when I went through the integration test with respect to this issue.

Comment (agajdosi) [reply test/integration/features/cmd-openshift.feature]: @LalatenduMohanty I have removed it previously as I have added docker container check before this step, tried this solution now again, but it does not help: https://ci.centos.org/job/minishift-pr/5535/consoleFull. So figuring out other solution.

**Comment by agajdosi** [comment]
Seems like "retrying oc status" catches the problem with the deployment of the application: https://ci.centos.org/job/minishift-pr/5539/consoleFull.

**Comment by agajdosi** [comment]
retest this please

**Comment by agajdosi** [comment]
@LalatenduMohanty This is probably ready, please take a look.

**Comment by LalatenduMohanty** [comment]
> @LalatenduMohanty This is probably ready, please take a look.

Sounds good. Lets see if it improves the situation.

**Pull request merged**